core/controllers/base.py

```diff
@@ -348,6 +348,9 @@ def render_template(
                 'Oppia is a free, open-source learning platform. Join the '
                 'community to create or try an exploration today!')
 
+        if 'nav_mode' not in values:
+            values['nav_mode'] = ''
```

Comment:
Moving the sidebar into its own directive split it off from the main base.html file. We generally don't use jinja in components, so I put nav_mode in GLOBALS and used that to indicate in the sidebar controller which menu option should be 'active'. Unfortunately, this led to some issues because not all pages have a nav_mode. So I had to introduce a default.

I'd appreciate your thoughts on this, though.